java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java

```diff
@@ -844,7 +844,7 @@ public void setValueCount(int valueCount) {
     }
     /* valueCount for the data vector is the current end offset */
     final int childValueCount = (valueCount == 0) ? 0 :
-            offsetBuffer.getInt((lastSet + 1) * OFFSET_WIDTH);
+            offsetBuffer.getInt(valueCount * OFFSET_WIDTH);
```

Comment:
shouldn't this be (valueCount + 1) * OFFSET_WIDTH?

Reply:
valueCount starts from 1

Reply:
yeah, good point :)

Reply:
Wait, scratch that. If you have one value, you need two offsets, right?

Reply:
yes, in that case valueCount will be 1 which gives the length of the first value = # values in data vector

Reply:
@jacques-n does this explanation make sense to you, what are the next steps here?

Reply:
I think the bug was actually that last set is not updated just above (fill the holes with offsets populates so more items are last set but doesn't actually update the last set value). If the vector was internally consistent, lastSet + 1 should be equal to valueCount. LastSet is zero-index and valuecount is 1-index so I believe the fix is correct. I think, probably that lastSet should also be updated at the end of the "fill the holes with offsets" section as well.